src/components/DataTable/DataTable.tsx

```diff
@@ -112,10 +123,13 @@ export function DataTable<TData, TValue>({
                     </DropdownMenuTrigger>
                     <DropdownMenuContent align="end">
                       <DropdownMenuLabel>Actions</DropdownMenuLabel>
-                      <ViewDialog row={row} />
-                      {EditDialog ? <EditDialog row={row} /> : null}
-                      {DeleteDialog ? <DeleteDialog row={row} /> : null}
-                      {RevokeKeyDialog ? <RevokeKeyDialog row={row} /> : null}
+                      <ViewDialog row={row as Row<RowType>} />
+                      {ContactEditDialog ? <ContactEditDialog row={row as Row<Contact>} /> : null}
+                      {TransactionEditDialog ? <TransactionEditDialog row={row as Row<Tx>} /> : null}
+                      {DestinationEditDialog ? <DestinationEditDialog row={row as Row<Destination>} /> : null}
+                      {DeleteDialog ? <DeleteDialog row={row as Row<Contact>} /> : null}
+                      {PaymailDeleteDialog ? <PaymailDeleteDialog row={row as Row<PaymailAddress>} /> : null}
+                      {RevokeKeyDialog ? <RevokeKeyDialog row={row as Row<AccessKey>} /> : null}
```

Comment:
ohhh it looks hmm.. not the best.

IMHO it would be much much better if this component would accept 2 optional components:
1. `inlineActions` which would define such buttons as Accept and Decline in contacts
2. `actions` which would define actions menu items and the logic how it should handle it.

With the proper way of using generics it should be easy to pass the row in a valid way to any action component.